packages/xrpl/src/sugar/submit.ts

```diff
@@ -122,10 +122,21 @@ async function waitForFinalTransactionOutcome(
 ): Promise<TxResponse> {
   await sleep(LEDGER_CLOSE_TIME)
 
-  const txResponse = await client.request({
-    command: 'tx',
-    transaction: txHash,
-  })
+  const txResponse = await client
+    .request({
+      command: 'tx',
+      transaction: txHash,
+    })
+    .catch(async (error) => {
+      // this type of error does not have its interface, type-assert and extract the value we need.
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions,@typescript-eslint/no-unsafe-member-access -- L131
+      const message = error.data.error as string
+      if (message === 'txnNotFound') {
+        return waitForFinalTransactionOutcome(client, txHash)
```

Comment:
This will never actually hit the condition where we check if we're past latestLedgerSequence.

Reply:
```
const txResponse = await client
    .request({
      command: 'tx',
      transaction: txHash,
    })
    .catch((error) => {
      if (message !== 'txnNotFound') {
            Promise.reject(error)
      }
      return null
    })

  if (txResponse != null && txResponse.result.validated) {
    return txResponse
  }
```

Reply:
If the transaction never gets validated it will just loop infinitely